pkg/transport/proxy/transparent/transparent_proxy.go

```diff
@@ -125,7 +130,52 @@ func (t *tracingTransport) forward(req *http.Request) (*http.Response, error) {
 	return tr.RoundTrip(req)
 }
 
+// manualForward manually forwards a request to the remote server using an HTTP client
+func (t *tracingTransport) manualForward(req *http.Request) (*http.Response, error) {
+	// Create a new request to the target URL
+	targetURL := t.p.targetURI + req.URL.Path
+	if req.URL.RawQuery != "" {
+		targetURL += "?" + req.URL.RawQuery
+	}
+
+	newReq, err := http.NewRequest(req.Method, targetURL, req.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new request: %w", err)
+	}
+
+	// Copy headers from the original request
+	for name, values := range req.Header {
+		for _, value := range values {
+			newReq.Header.Add(name, value)
+		}
+	}
+
+	// Create HTTP client and make the request
+	client := &http.Client{
+		Timeout: 30 * time.Second,
+	}
+
+	return client.Do(newReq)
```

Comment:
this HTTP client doesn't use authentication at all... how is this used? I'm a little confused by this tbh.